### PR TITLE
try out some new beta docker travis features

### PR DESCRIPTION
**Pull request by reaperhulk**

### CODE REVIEW

File: .travis/install.sh [hunk 1]
```diff
@@ -3,6 +3,7 @@
 set -e
 set -x
 
+export PATH="~/.local/bin;$PATH"
```
Comment (alex) [inline .travis/install.sh]: This should be a colon, not a semi-colon right?

Comment (reaperhulk) [inline .travis/install.sh]: doh!

**Comment by alex** [comment]
retest this please

File: .travis.yml [hunk 2]
```diff
@@ -1,3 +1,22 @@
+sudo: false
+addons:
+  apt_sources:
```
Comment (alex) [inline .travis.yml]: Nit, but the indent for this section is inconsistent witht he rest of the file

**Comment by alex** [comment]
merge conflicts here :-(

**Comment by alex** [comment]
(Maybe wait until I confirm that the addon is enabled for PRs though)

File: .travis/install.sh [hunk 3]
```diff
@@ -51,54 +52,13 @@ if [[ "$(uname -s)" == 'Darwin' ]]; then
...
-            ;;
-        docs)
-            sudo apt-get install libenchant-dev
-            ;;
-    esac
 fi
 
-sudo pip install virtualenv
+pip install --user virtualenv
 virtualenv ~/.venv
```
Comment (alex) [inline .travis/install.sh]: What do you think about using `python -m virtualenv ~/.venv` so we don't need to munge `$PATH`?

**Comment by alex** [comment]
It looks like the travis changes will go live on Monday

**Comment by alex** [comment]
@reaperhulk the structure changed a bit here: https://github.com/meatballhat/yolo-octo-adventure/commit/ec82dc725ea8f1d326abfba75908d38aaf85dd1e but it should be live on PRs

File: .travis.yml [hunk 4]
```diff
@@ -5,6 +5,26 @@ os:
 compiler:
     - clang
     - gcc
+sudo: false
+addons:
+    apt:
+        sources::
```
Comment (alex) [inline .travis.yml]: Err, extra colon?

Comment (reaperhulk) [inline .travis.yml]: son of a... surprised that's legal yaml.

**Comment by alex** [comment]
@meatballhat is investigating why this doesn't seem to be working on PRs

**Comment by alex** [comment]
It looks like the diciontary the docs tests are running against has fewer words, or something, we'll need to update for that.

**Comment by alex** [comment]
Hmm. Why are we losing coverage?

**Comment by alex** [comment]
Looks like maybe https://coveralls.io/builds/2229276/source?filename=src%2Fcryptography%2Fhazmat%2Fbackends%2Fopenssl%2Fbackend.py#L981 is the answer

**Comment by alex** [comment]
Also https://coveralls.io/builds/2229276/source?filename=tests%2Fhazmat%2Fprimitives%2Ftest_ec.py#L53 ?

**Comment by alex** [comment]
0.9.8 stuff isn't actually working because it doesn't isntall lucid/opensls-dev with `--force`. Pending a pony from @meatballhat :-)

File: .travis.yml [hunk 5]
```diff
@@ -1,29 +1,115 @@
...
-        - TOXENV=pep8
-        - TOXENV=py3pep8
+sudo: false
+language: python
+cache:
+    directories:
+        - $HOME/.cache/pip
+matrix:
+    include:
+        - python: 2.6 # these are just to make travis's UI a bit prettier
```
Comment (alex) [inline .travis.yml]: Nit, but can you put the comment above the line? I think we're using that style everywhere else.

Also, I don't see the linux PyPy builds anymore?

File: .travis/install.sh [hunk 6]
```diff
@@ -51,55 +50,15 @@ if [[ "$(uname -s)" == 'Darwin' ]]; then
...
-        docs)
-            sudo apt-get install libenchant-dev
-            ;;
-    esac
+    pip install virtualenv
 fi
 
-sudo pip install virtualenv
-virtualenv ~/.venv
+python -m virtualenv ~/.venv
```
Comment (alex) [inline .travis/install.sh]: Is it necessary to create a new venv on linux?

File: docs/spelling_wordlist.txt [hunk 7]
```diff
@@ -36,6 +36,7 @@ iterable
 Koblitz
 Lange
 metadata
+multi
```
Comment (alex) [inline docs/spelling_wordlist.txt]: ?

File: .travis.yml (hunk 5 above)
Comment (reaperhulk) [inline .travis.yml]: Yeah...looks like I screwed up and dropped that.

File: .travis/install.sh (hunk 6 above)
Comment (reaperhulk) [inline .travis/install.sh]: No, but it simplifies things by making it so we don't have to check if we're in darwin or not to determine if we need to source into this venv.

File: docs/spelling_wordlist.txt (hunk 7 above)
Comment (reaperhulk) [inline docs/spelling_wordlist.txt]: In docker infra the "multi-threaded" line in the changelog threw a spelling error.

File: .travis/install.sh (hunk 6 above)
Comment (alex) [inline .travis/install.sh]: k

On Mon, Apr 13, 2015 at 4:43 PM, Paul Kehrer <notifications@github.com>
wrote:

> In .travis/install.sh
> <https://github.com/pyca/cryptography/pull/1785#discussion_r28277071>:
>
> >  fi
> >
> > -sudo pip install virtualenv
> > -virtualenv ~/.venv
> > +python -m virtualenv ~/.venv
>
> No, but it simplifies things by making it so we don't have to check if
> we're in darwin or not to determine if we need to source into this venv.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1785/files#r28277071>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084

File: docs/spelling_wordlist.txt (hunk 7 above)
Comment (alex) [inline docs/spelling_wordlist.txt]: k

On Mon, Apr 13, 2015 at 4:43 PM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/spelling_wordlist.txt
> <https://github.com/pyca/cryptography/pull/1785#discussion_r28277095>:
>
> > @@ -36,6 +36,7 @@ iterable
> >  Koblitz
> >  Lange
> >  metadata
> > +multi
>
> In docker infra the "multi-threaded" line in the changelog threw a
> spelling error.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1785/files#r28277095>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084

**Pull request merged**